## Personal Project - Fairuz Idris
### Analysing COVID-19 cases using SQL 
Database is obtained from kaggle.com

![](erd.png)

#### ERD diagram is established using Lucidchart 

In [1]:
pip install ipython-sql

In [2]:
%load_ext sql

In [3]:
%sql sqlite:///database.db

In [4]:
%sql SELECT sql from sqlite_master WHERE type='table';

 * sqlite:///database.db
Done.


[('CREATE TABLE data (time_id INTEGER,country_id INTEGER,cases INTEGER,deaths INTEGER,recovered INTEGER,cumulate_cases INTEGER,cumulate_deaths INTEGER,cumulate_recovered INTEGER,source_id INTEGER)',),
 ('CREATE TABLE DimTime (id INTEGER,date DateTime,day INTEGER,month INTEGER,year INTEGER,cw INTEGER,day_of_week INTEGER)',),
 ('CREATE TABLE DimCountry (name,country_code,continent,population,convert_name)',),
 ('CREATE TABLE DimSource (id INTEGER PRIMARY KEY AUTOINCREMENT,name TEXT)',),
 ('CREATE TABLE sqlite_sequence(name,seq)',),
 ('CREATE TABLE Measure (country_id,time_id,value,note,category,source_id)',),
 ('CREATE TABLE CountryHealthIndex (country_id,time_id,value)',),
 ('CREATE TABLE CountryTestingPolicy (country_id,time_id,is_testing)',)]

#### Let's see the entity for data

In [4]:
%%sql SELECT * FROM data
LIMIT 100;

 * sqlite:///database.db
Done.


[(758, 1, 0, 0, None, 0, 0, None, 3),
 (759, 1, 0, 0, None, 0, 0, None, 3),
 (760, 1, 0, 0, None, 0, 0, None, 3),
 (761, 1, 0, 0, None, 0, 0, None, 3),
 (762, 1, 0, 0, None, 0, 0, None, 3),
 (763, 1, 0, 0, None, 0, 0, None, 3),
 (764, 1, 0, 0, None, 0, 0, None, 3),
 (765, 1, 0, 0, None, 0, 0, None, 3),
 (766, 1, 0, 0, None, 0, 0, None, 3),
 (767, 1, 0, 0, None, 0, 0, None, 3),
 (768, 1, 0, 0, None, 0, 0, None, 3),
 (769, 1, 0, 0, None, 0, 0, None, 3),
 (770, 1, 0, 0, None, 0, 0, None, 3),
 (771, 1, 0, 0, None, 0, 0, None, 3),
 (772, 1, 0, 0, None, 0, 0, None, 3),
 (773, 1, 0, 0, None, 0, 0, None, 3),
 (774, 1, 0, 0, None, 0, 0, None, 3),
 (775, 1, 0, 0, None, 0, 0, None, 3),
 (776, 1, 0, 0, None, 0, 0, None, 3),
 (777, 1, 0, 0, None, 0, 0, None, 3),
 (778, 1, 0, 0, None, 0, 0, None, 3),
 (779, 1, 0, 0, None, 0, 0, None, 3),
 (780, 1, 0, 0, None, 0, 0, None, 3),
 (781, 1, 0, 0, None, 0, 0, None, 3),
 (782, 1, 0, 0, None, 0, 0, None, 3),
 (783, 1, 0, 0, None, 0, 0, None, 3),
 (784, 1, 0, 0, None, 0, 0, None, 3),
 (785, 1, 0, 0, None, 0, 0, None, 3),
 (786, 1, 0, 0, None, 0, 0, None, 3),
 (787, 1, 0, 0, None, 0, 0, None, 3),
 (788, 1, 0, 0, None, 0, 0, None, 3),
 (789, 1, 0, 0, None, 0, 0, None, 3),
 (790, 1, 0, 0, None, 0, 0, None, 3),
 (791, 1, 0, 0, None, 0, 0, None, 3),
 (792, 1, 0, 0, None, 0, 0, None, 3),
 (793, 1, 0, 0, None, 0, 0, None, 3),
 (794, 1, 0, 0, None, 0, 0, None, 3),
 (795, 1, 0, 0, None, 0, 0, None, 3),
 (796, 1, 0, 0, None, 0, 0, None, 3),
 (797, 1, 0, 0, None, 0, 0, None, 3),
 (798, 1, 0, 0, None, 0, 0, None, 3),
 (799, 1, 0, 0, None, 0, 0, None, 3),
 (800, 1, 0, 0, None, 0, 0, None, 3),
 (801, 1, 0, 0, None, 0, 0, None, 3),
 (802, 1, 0, 0, None, 0, 0, None, 3),
 (803, 1, 0, 0, None, 0, 0, None, 3),
 (804, 1, 0, 0, None, 0, 0, None, 3),
 (805, 1, 0, 0, None, 0, 0, None, 3),
 (806, 1, 0, 0, None, 0, 0, None, 3),
 (807, 1, 0, 0, None, 0, 0, None, 3),
 (808, 1, 0, 0, None, 0, 0, None, 3),
 (809, 1, 0, 0, None, 0, 0, None, 3),
 (810, 1, 0, 0, None, 0, 0, None, 3),
 (811, 1, 0, 0, None, 0, 0, None, 3),
 (812, 1, 0, 0, None, 0, 0, None, 3),
 (813, 1, 0, 0, None, 0, 0, None, 3),
 (814, 1, 1, 0, None, 1, 0, None, 3),
 (815, 1, 0, 0, None, 1, 0, None, 3),
 (816, 1, 0, 0, None, 1, 0, None, 3),
 (817, 1, 0, 0, None, 1, 0, None, 3),
 (818, 1, 0, 0, None, 1, 0, None, 3),
 (819, 1, 0, 0, None, 1, 0, None, 3),
 (820, 1, 0, 0, None, 1, 0, None, 3),
 (826, 1, 3, 0, None, 4, 0, None, 3),
 (829, 1, 3, 0, None, 7, 0, None, 3),
 (833, 1, 3, 0, None, 10, 0, None, 3),
 (834, 1, 6, 0, None, 16, 0, None, 3),
 (835, 1, 5, 0, None, 21, 0, None, 3),
 (836, 1, 1, 0, None, 22, 0, None, 3),
 (837, 1, 0, 0, None, 22, 0, None, 3),
 (838, 1, 0, 0, None, 22, 0, None, 3),
 (839, 1, 2, 0, None, 24, 0, None, 3),
 (840, 1, 0, 0, None, 24, 0, None, 3),
 (841, 1, 10, 0, None, 34, 0, None, 3),
 (842, 1, 6, 1, None, 40, 1, None, 3),
 (843, 1, 2, 0, None, 42, 1, None, 3),
 (844, 1, 33, 0, None, 75, 1, None, 3),
 (845, 1, 0, 0, None, 75, 1, None, 3),
 (846, 1, 16, 1, None, 91, 2, None, 3),
 (847, 1, 15, 1, None, 106, 3, None, 3),
 (848, 1, 8, 1, None, 114, 4, None, 3),
 (849, 1, 27, 0, None, 141, 4, None, 3),
 (850, 1, 25, 0, None, 166, 4, None, 3),
 (851, 1, 26, 0, None, 192, 4, None, 3),
 (852, 1, 43, 0, None, 235, 4, None, 3),
 (853, 1, 0, 0, None, 235, 4, None, 3),
 (854, 1, 35, 1, None, 270, 5, None, 3),
 (855, 1, 29, 2, None, 299, 7, None, 3),
 (856, 1, 38, 0, None, 337, 7, None, 3),
 (857, 1, 30, 4, None, 367, 11, None, 3),
 (858, 1, 56, 3, None, 423, 14, None, 3),
 (859, 1, 61, 1, None, 484, 15, None, 3),
 (860, 1, 37, 0, None, 521, 15, None, 3),
 (861, 1, 34, 3, None, 555, 18, None, 3),
 (862, 1, 52, 0, None, 607, 18, None, 3),
 (863, 1, 58, 3, None, 665, 21, None, 3),
 (864, 1, 49, 2, None, 714, 23, None, 3),
 (865, 1, 70, 2, None, 784, 25, None, 3),
 (866, 1, 10, 4, None, 794, 29, None, 3),
 (867, 1, 51, 1, None, 845, 30, None, 3)]

#### Source of data for each country

In [11]:
%%sql SELECT 
    d.country_id,
    d.cases,
    d.deaths,
    d.recovered,
    d.source_id,
    ds.name
FROM Data d
JOIN DimSource ds
    ON d.source_id = ds.id
LIMIT 200;

 * sqlite:///database.db
Done.


[(1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 1, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 3, 0, None, 3, 'WHO'),
 (1, 3, 0, None, 3, 'WHO'),
 (1, 3, 0, None, 3, 'WHO'),
 (1, 6, 0, None, 3, 'WHO'),
 (1, 5, 0, None, 3, 'WHO'),
 (1, 1, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 2, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 10, 0, None, 3, 'WHO'),
 (1, 6, 1, None, 3, 'WHO'),
 (1, 2, 0, None, 3, 'WHO'),
 (1, 33, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 16, 1, None, 3, 'WHO'),
 (1, 15, 1, None, 3, 'WHO'),
 (1, 8, 1, None, 3, 'WHO'),
 (1, 27, 0, None, 3, 'WHO'),
 (1, 25, 0, None, 3, 'WHO'),
 (1, 26, 0, None, 3, 'WHO'),
 (1, 43, 0, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 35, 1, None, 3, 'WHO'),
 (1, 29, 2, None, 3, 'WHO'),
 (1, 38, 0, None, 3, 'WHO'),
 (1, 30, 4, None, 3, 'WHO'),
 (1, 56, 3, None, 3, 'WHO'),
 (1, 61, 1, None, 3, 'WHO'),
 (1, 37, 0, None, 3, 'WHO'),
 (1, 34, 3, None, 3, 'WHO'),
 (1, 52, 0, None, 3, 'WHO'),
 (1, 58, 3, None, 3, 'WHO'),
 (1, 49, 2, None, 3, 'WHO'),
 (1, 70, 2, None, 3, 'WHO'),
 (1, 10, 4, None, 3, 'WHO'),
 (1, 51, 1, None, 3, 'WHO'),
 (1, 63, 0, None, 3, 'WHO'),
 (1, 88, 3, None, 3, 'WHO'),
 (1, 35, 2, None, 3, 'WHO'),
 (1, 61, 1, None, 3, 'WHO'),
 (1, 84, 4, None, 3, 'WHO'),
 (1, 105, 2, None, 3, 'WHO'),
 (1, 70, 1, None, 3, 'WHO'),
 (1, 112, 4, None, 3, 'WHO'),
 (1, 68, 10, None, 3, 'WHO'),
 (1, 172, 0, None, 3, 'WHO'),
 (1, 124, 3, None, 3, 'WHO'),
 (1, 122, 0, None, 3, 'WHO'),
 (1, 222, 4, None, 3, 'WHO'),
 (1, 164, 4, None, 3, 'WHO'),
 (1, 134, 4, None, 3, 'WHO'),
 (1, 235, 13, None, 3, 'WHO'),
 (1, 190, 5, None, 3, 'WHO'),
 (1, 330, 5, None, 3, 'WHO'),
 (1, 168, 9, None, 3, 'WHO'),
 (1, 171, 2, None, 3, 'WHO'),
 (1, 215, 3, None, 3, 'WHO'),
 (1, 255, 6, None, 3, 'WHO'),
 (1, 369, 5, None, 3, 'WHO'),
 (1, 285, 2, None, 3, 'WHO'),
 (1, 280, 5, None, 3, 'WHO'),
 (1, 259, 3, None, 3, 'WHO'),
 (1, 113, 6, None, 3, 'WHO'),
 (1, 1063, 32, None, 3, 'WHO'),
 (1, 0, 0, None, 3, 'WHO'),
 (1, 262, 1, None, 3, 'WHO'),
 (1, 408, 4, None, 3, 'WHO'),
 (1, 581, 5, None, 3, 'WHO'),
 (1, 492, 9, None, 3, 'WHO'),
 (1, 531, 6, None, 3, 'WHO'),
 (1, 540, 12, None, 3, 'WHO'),
 (1, 782, 11, None, 3, 'WHO'),
 (1, 584, 2, None, 3, 'WHO'),
 (1, 591, 1, None, 3, 'WHO'),
 (1, 658, 1, None, 3, 'WHO'),
 (1, 625, 7

source id 1 = ECDC,
source id 2 = JHU,
source id 3 = WHO 

#### Total number of countries that are taking into account

In [9]:
%%sql SELECT COUNT (DISTINCT country_id)
FROM data

 * sqlite:///database.db
Done.


[(210,)]

#### Total number of cases for 2019 and 2020 across 210 countries

In [15]:
%%sql SELECT sum(cases)
FROM data

 * sqlite:///database.db
Done.


[(30002042,)]

#### Country with the highest cases

In [12]:
%%sql SELECT country_id, cumulate_cases
FROM data
ORDER BY cumulate_cases DESC
LIMIT 1;

 * sqlite:///database.db
Done.


[(201, 2548996)]

#### Country with the highest deaths

In [13]:
%%sql SELECT country_id, cumulate_deaths
FROM data
ORDER BY cumulate_deaths DESC
LIMIT 1;

 * sqlite:///database.db
Done.


[(201, 125804)]

#### Country with the highest recovery 

In [14]:
%%sql SELECT country_id, cumulate_recovered
FROM data
ORDER BY cumulate_recovered DESC
LIMIT 1;

 * sqlite:///database.db
Done.


[(28, 746018)]

### Analysing country with the highest cases (Country id = 201)

#### Let's see the entity for DimTime

In [16]:
%%sql SELECT * FROM DimTime
ORDER BY id DESC
LIMIT 100;

 * sqlite:///database.db
Done.


[(1124, '2020-12-31', 31, 12, 2020, 53, 4),
 (1123, '2020-12-30', 30, 12, 2020, 53, 3),
 (1122, '2020-12-29', 29, 12, 2020, 53, 2),
 (1121, '2020-12-28', 28, 12, 2020, 53, 1),
 (1120, '2020-12-27', 27, 12, 2020, 53, 7),
 (1119, '2020-12-26', 26, 12, 2020, 52, 6),
 (1118, '2020-12-25', 25, 12, 2020, 52, 5),
 (1117, '2020-12-24', 24, 12, 2020, 52, 4),
 (1116, '2020-12-23', 23, 12, 2020, 52, 3),
 (1115, '2020-12-22', 22, 12, 2020, 52, 2),
 (1114, '2020-12-21', 21, 12, 2020, 52, 1),
 (1113, '2020-12-20', 20, 12, 2020, 52, 7),
 (1112, '2020-12-19', 19, 12, 2020, 51, 6),
 (1111, '2020-12-18', 18, 12, 2020, 51, 5),
 (1110, '2020-12-17', 17, 12, 2020, 51, 4),
 (1109, '2020-12-16', 16, 12, 2020, 51, 3),
 (1108, '2020-12-15', 15, 12, 2020, 51, 2),
 (1107, '2020-12-14', 14, 12, 2020, 51, 1),
 (1106, '2020-12-13', 13, 12, 2020, 51, 7),
 (1105, '2020-12-12', 12, 12, 2020, 50, 6),
 (1104, '2020-12-11', 11, 12, 2020, 50, 5),
 (1103, '2020-12-10', 10, 12, 2020, 50, 4),
 (1102, '2020-12-09', 9, 12, 2020, 50, 3),
 (1101, '2020-12-08', 8, 12, 2020, 50, 2),
 (1100, '2020-12-07', 7, 12, 2020, 50, 1),
 (1099, '2020-12-06', 6, 12, 2020, 50, 7),
 (1098, '2020-12-05', 5, 12, 2020, 49, 6),
 (1097, '2020-12-04', 4, 12, 2020, 49, 5),
 (1096, '2020-12-03', 3, 12, 2020, 49, 4),
 (1095, '2020-12-02', 2, 12, 2020, 49, 3),
 (1094, '2020-12-01', 1, 12, 2020, 49, 2),
 (1093, '2020-11-30', 30, 11, 2020, 49, 1),
 (1092, '2020-11-29', 29, 11, 2020, 49, 7),
 (1091, '2020-11-28', 28, 11, 2020, 48, 6),
 (1090, '2020-11-27', 27, 11, 2020, 48, 5),
 (1089, '2020-11-26', 26, 11, 2020, 48, 4),
 (1088, '2020-11-25', 25, 11, 2020, 48, 3),
 (1087, '2020-11-24', 24, 11, 2020, 48, 2),
 (1086, '2020-11-23', 23, 11, 2020, 48, 1),
 (1085, '2020-11-22', 22, 11, 2020, 48, 7),
 (1084, '2020-11-21', 21, 11, 2020, 47, 6),
 (1083, '2020-11-20', 20, 11, 2020, 47, 5),
 (1082, '2020-11-19', 19, 11, 2020, 47, 4),
 (1081, '2020-11-18', 18, 11, 2020, 47, 3),
 (1080, '2020-11-17', 17, 11, 2020, 47, 2),
 (1079, '2020-11-16', 16, 11, 2020, 47, 1),
 (1078, '2020-11-15', 15, 11, 2020, 47, 7),
 (1077, '2020-11-14', 14, 11, 2020, 46, 6),
 (1076, '2020-11-13', 13, 11, 2020, 46, 5),
 (1075, '2020-11-12', 12, 11, 2020, 46, 4),
 (1074, '2020-11-11', 11, 11, 2020, 46, 3),
 (1073, '2020-11-10', 10, 11, 2020, 46, 2),
 (1072, '2020-11-09', 9, 11, 2020, 46, 1),
 (1071, '2020-11-08', 8, 11, 2020, 46, 7),
 (1070, '2020-11-07', 7, 11, 2020, 45, 6),
 (1069, '2020-11-06', 6, 11, 2020, 45, 5),
 (1068, '2020-11-05', 5, 11, 2020, 45, 4),
 (1067, '2020-11-04', 4, 11, 2020, 45, 3),
 (1066, '2020-11-03', 3, 11, 2020, 45, 2),
 (1065, '2020-11-02', 2, 11, 2020, 45, 1),
 (1064, '2020-11-01', 1, 11, 2020, 45, 7),
 (1063, '2020-10-31', 31, 10, 2020, 44, 6),
 (1062, '2020-10-30', 30, 10, 2020, 44, 5),
 (1061, '2020-10-29', 29, 10, 2020, 44, 4),
 (1060, '2020-10-28', 28, 10, 2020, 44, 3),
 (1059, '2020-10-27', 27, 10, 2020, 44, 2),
 (1058, '2020-10-26', 26, 10, 2020, 44, 1),
 (1057, '2020-10-25', 25, 10, 2020, 44, 7),
 (1056, '2020-10-24', 24, 10, 2020, 43, 6),
 (1055, '2020-10-23', 23, 10, 2020, 43, 5),
 (1054, '2020-10-22', 22, 10, 2020, 43, 4),
 (1053, '2020-10-21', 21, 10, 2020, 43, 3),
 (1052, '2020-10-20', 20, 10, 2020, 43, 2),
 (1051, '2020-10-19', 19, 10, 2020, 43, 1),
 (1050, '2020-10-18', 18, 10, 2020, 43, 7),
 (1049, '2020-10-17', 17, 10, 2020, 42, 6),
 (1048, '2020-10-16', 16, 10, 2020, 42, 5),
 (1047, '2020-10-15', 15, 10, 2020, 42, 4),
 (1046, '2020-10-14', 14, 10, 2020, 42, 3),
 (1045, '2020-10-13', 13, 10, 2020, 42, 2),
 (1044, '2020-10-12', 12, 10, 2020, 42, 1),
 (1043, '2020-10-11', 11, 10, 2020, 42, 7),
 (1042, '2020-10-10', 10, 10, 2020, 41, 6),
 (1041, '2020-10-09', 9, 10, 2020, 41, 5),
 (1040, '2020-10-08', 8, 10, 2020, 41, 4),
 (1039, '2020-10-07', 7, 10, 2020, 41, 3),
 (1038, '2020-10-06', 6, 10, 2020, 41, 2),
 (1037, '2020-10-05', 5, 10, 2020, 41, 1),
 (1036, '2020-10-04', 4, 10, 2020, 41, 7),
 (1035, '2020-10-03', 3, 10, 2020, 40, 6),
 (1034, '2020-10-02', 2, 10, 2020, 40, 5),
 (1033, '2020-10-01', 

#### Cases, deaths and recovery for year 2019 and 2020 based on country 

In [20]:
%%sql SELECT 
    d.country_id,
    d.cumulate_cases,
    d.cumulate_deaths,
    d.cumulate_recovered,
    dt.month,
    dt.year  
FROM Data d
JOIN DimTime dt
    ON d.time_id = dt.id
ORDER BY country_id
LIMIT 200;

 * sqlite:///database.db
Done.


[(1, 0, 0, None, 12, 2019),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 1, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 0, 0, None, 2, 2020),
 (1, 1, 0, None, 2, 2020),
 (1, 1, 0, None, 2, 2020),
 (1, 1, 0, None, 2, 2020),
 (1, 1, 0, None, 2, 2020),
 (1, 1, 0, None, 2, 2020),
 (1, 1, 0, None, 3, 2020),
 (1, 1, 0, None, 3, 2020),
 (1, 4, 0, None, 3, 2020),
 (1, 7, 0, None, 3, 2020),
 (1, 10, 0, None, 3, 2020),
 (1, 16, 0, None, 3, 2020),
 (1, 21, 0, None, 3, 2020),
 (1, 22, 0, None, 3, 2020),
 (1, 22, 0, None, 3, 2020),
 (1, 22, 0, None, 3, 2020),
 (1, 24, 0, None, 3, 2020),
 (1, 24, 0, None, 3, 2020),
 (1, 34, 0, None, 3, 2020),
 (1, 40, 1, None, 3, 2020),
 (1, 42, 1, None, 3, 2020),
 (1, 75, 1, None, 3, 2020),
 (1, 75, 1, None, 3, 2020),
 (1, 91, 2, None, 3, 2020),
 (1, 106, 3, None, 3, 2020),
 (1, 114, 4, None, 3, 2020),
 (1, 141, 4, None, 3, 2020),
 (1, 166, 4, None, 4, 2020),
 (1, 192, 4, None, 4, 2020),
 (1, 235, 4, None, 4, 2020),
 (1, 235, 4, None, 4, 2020),
 (1, 270, 5, None, 4, 2020),
 (1, 299, 7, None, 4, 2020),
 (1, 337, 7, None, 4, 2020),
 (1, 367, 11, None, 4, 2020),
 (1, 423, 14, None, 4, 2020),
 (1, 484, 15, None, 4, 2020),
 (1, 521, 15, None, 4, 2020),
 (1, 555, 18, None, 4, 2020),
 (1, 607, 18, None, 4, 2020),
 (1, 665, 21, None, 4, 2020),
 (1, 714, 23, None, 4, 2020),
 (1, 784, 25, None, 4, 2020),
 (1, 794, 29, None, 4, 2020),
 (1, 845, 30, None, 4, 2020),
 (1, 908, 30, None, 4, 2020),
 (1, 996, 33, None, 4, 2020),
 (1, 1031, 35, None, 4, 2020),
 (1, 1092, 36, None, 4, 2020),
 (1, 1176, 40, None, 4, 2020),
 (1, 1281, 42, None, 4, 2020),
 (1, 1351, 43, None, 4, 2020),
 (1, 1463, 47, None, 4, 2020),
 (1, 1531, 57, None, 4, 2020),
 (1, 1703, 57, None, 4, 2020),
 (1, 1827, 60, None, 4, 2020),
 (1, 1949, 60, None, 4, 2020),
 (1, 2171, 64, None, 5, 2020),
 (1, 2335, 68, None, 5, 2020),
 (1, 2469, 72, None, 5, 2020),
 (1, 2704, 85, None, 5, 2020),
 (1, 2894, 90, None, 5, 2020),
 (1, 3224, 95, None, 5, 2020),
 (1, 3392, 104, None, 5, 2020),
 (1, 3563, 106, None, 5, 2020),
 (1, 3778, 109, None, 5, 2020),
 (1, 4033, 115, None, 5, 2020),
 (1, 4402, 120, None, 5, 2020),
 (1, 4687, 122, None, 5, 2020),
 (1, 4967, 127, None, 5, 2020),
 (1, 5226, 130, None, 5, 2020),
 (1, 5339, 136, None, 5, 2020),
 (1, 6402, 168, None, 5, 2020),
 (1, 6402, 168, None, 5, 2020),
 (1, 6664, 169, None, 5, 2020),
 (1, 7072, 173, None, 5, 2020),
 (1, 7653, 178, None, 5, 2020),
 (1, 8145, 187, None, 5, 2020),
 (1, 8676, 193, None, 5, 2020),
 (1, 9216, 205, None, 5, 2020),
 (1, 9998, 216, None, 5, 2020),
 (1, 10582, 218, None, 5, 2020),
 (1, 11173, 219, None, 5, 2020),
 (1, 11831, 220, None, 5, 2020),
 

#### Number of cases for each day 

In [19]:
%%sql SELECT
    d.time_id,
    d.country_id,
    d.cases,
    dt.date
FROM Data d
JOIN Dimtime dt
    ON d.time_id = dt.id
WHERE country_id = 201
LIMIT 200;

 * sqlite:///database.db
Done.


[(758, 201, 0, '2019-12-31'),
 (759, 201, 0, '2020-01-01'),
 (760, 201, 0, '2020-01-02'),
 (761, 201, 0, '2020-01-03'),
 (762, 201, 0, '2020-01-04'),
 (763, 201, 0, '2020-01-05'),
 (764, 201, 0, '2020-01-06'),
 (765, 201, 0, '2020-01-07'),
 (766, 201, 0, '2020-01-08'),
 (767, 201, 0, '2020-01-09'),
 (768, 201, 0, '2020-01-10'),
 (769, 201, 0, '2020-01-11'),
 (770, 201, 0, '2020-01-12'),
 (771, 201, 0, '2020-01-13'),
 (772, 201, 0, '2020-01-14'),
 (773, 201, 0, '2020-01-15'),
 (774, 201, 0, '2020-01-16'),
 (775, 201, 0, '2020-01-17'),
 (776, 201, 0, '2020-01-18'),
 (777, 201, 0, '2020-01-19'),
 (778, 201, 0, '2020-01-20'),
 (779, 201, 1, '2020-01-21'),
 (780, 201, 0, '2020-01-22'),
 (781, 201, 0, '2020-01-23'),
 (782, 201, 0, '2020-01-24'),
 (783, 201, 1, '2020-01-25'),
 (784, 201, 0, '2020-01-26'),
 (785, 201, 3, '2020-01-27'),
 (786, 201, 0, '2020-01-28'),
 (787, 201, 0, '2020-01-29'),
 (788, 201, 0, '2020-01-30'),
 (789, 201, 1, '2020-01-31'),
 (790, 201, 1, '2020-02-01'),
 (791, 201, 1, '2020-02-02'),
 (792, 201, 3, '2020-02-03'),
 (793, 201, 0, '2020-02-04'),
 (794, 201, 0, '2020-02-05'),
 (795, 201, 1, '2020-02-06'),
 (796, 201, 0, '2020-02-07'),
 (797, 201, 0, '2020-02-08'),
 (798, 201, 0, '2020-02-09'),
 (799, 201, 0, '2020-02-10'),
 (800, 201, 1, '2020-02-11'),
 (801, 201, 0, '2020-02-12'),
 (802, 201, 1, '2020-02-13'),
 (803, 201, 1, '2020-02-14'),
 (804, 201, 0, '2020-02-15'),
 (805, 201, 0, '2020-02-16'),
 (806, 201, 0, '2020-02-17'),
 (807, 201, 0, '2020-02-18'),
 (808, 201, 0, '2020-02-19'),
 (809, 201, 0, '2020-02-20'),
 (810, 201, 1, '2020-02-21'),
 (811, 201, 19, '2020-02-22'),
 (812, 201, 0, '2020-02-23'),
 (813, 201, 0, '2020-02-24'),
 (814, 201, 18, '2020-02-25'),
 (815, 201, 0, '2020-02-26'),
 (816, 201, 6, '2020-02-27'),
 (817, 201, 1, '2020-02-28'),
 (818, 201, 6, '2020-02-29'),
 (819, 201, 3, '2020-03-01'),
 (820, 201, 20, '2020-03-02'),
 (821, 201, 14, '2020-03-03'),
 (822, 201, 22, '2020-03-04'),
 (823, 201, 34, '2020-03-05'),
 (824, 201, 74, '2020-03-06'),
 (825, 201, 105, '2020-03-07'),
 (826, 201, 95, '2020-03-08'),
 (827, 201, 121, '2020-03-09'),
 (828, 201, 200, '2020-03-10'),
 (829, 201, 271, '2020-03-11'),
 (830, 201, 287, '2020-03-12'),
 (831, 201, 351, '2020-03-13'),
 (832, 201, 511, '2020-03-14'),
 (833, 201, 777, '2020-03-15'),
 (834, 201, 823, '2020-03-16'),
 (835, 201, 887, '2020-03-17'),
 (836, 201, 1766, '2020-03-18'),
 (837, 201, 2988, '2020-03-19'),
 (838, 201, 4835, '2020-03-20'),
 (839, 201, 5374, '2020-03-21'),
 (840, 201, 7123, '2020-03-22'),
 (841, 201, 8459, '2020-03-23'),
 (842, 201, 11236, '2020-03-24'),
 (843, 201, 8789, '2020-03-25'),
 (844, 201, 13963, '2020-03-26'),
 (845, 201, 16797, '2020-03-27'),
 (846, 201, 18695, '2020-03-28'),
 (847, 201, 19979, '2020-03-29'),
 (848, 201, 18360, '2020-03-30'),
 (849, 201, 21595, '2020-03-31'),
 (850, 201, 24998, '2020-04-01'),
 (851, 201, 27103, '2020-04-02'),
 (852, 201, 28819, '2020-04-03'),
 (853, 201, 32425, '2020-04-04'),
 (854, 201, 34272, '2020-04-05'),
 (855, 201, 25398, '2020-04-06'),
 (856, 201, 30561, '2020-04-07'),
 (857, 201, 30613, '2020-04-08'),
 (858, 201, 33323, '2020-04-09'),
 (859, 201, 33901, '2020-04-10'),
 (860, 201, 35527, '2020-04-11'),
 (861, 201, 28391, '2020-04-12'),
 (862, 201, 27620, '2020-04-13'),
 (863, 201, 25023, '2020-04-14'),
 (864, 201, 26922, '2020-04-15'),
 (865, 201, 30148, '2020-04-16'),
 (866, 201, 31667, '2020-04-17'),
 (867, 201, 30833, '2020-04-18'),
 (868, 201, 32922, '2020-04-19'),
 (869, 201, 24601, '2020-04-20'),
 (870, 201, 28065, '2020-04-21'),
 (871, 201, 37289, '2020-04-22'),
 (872, 201, 17588, '2020-04-23'),
 (873, 201, 26543, '2020-04-24'),
 (874, 201, 21352, '2020-04-25'),
 (875, 201, 48529, '2020-04-26'),
 (876, 201, 26857, '2020-04-27'),
 (877, 201, 22541, '2020-04-28'),
 (878, 201, 24132, '2020-04-29'),
 (879, 201, 27326, '2020-04-30'),
 (880, 201, 29917, '2020-05-01'),
 (881, 201, 33955, '2020-05-02'),
 (882, 201, 29288, '2020-05-03'),
 (883, 201, 24972, '2020-05-04'),
 (884, 

#### Peak cases at the country 

In [26]:
%%sql SELECT
    d.time_id,
    d.cases,
    dt.date,
    m.category,
    m.value
FROM Data d
JOIN Dimtime dt
    ON d.time_id = dt.id
JOIN Measure m
    ON d.country_id = m.country_id
WHERE d.country_id = 201
ORDER BY d.cases DESC
LIMIT 1; 

 * sqlite:///database.db
Done.


[(875, 48529, '2020-04-26', 'Movement restrictions', 'Visa restrictions')]

#### Covid-19 transmission and protective measures taken by the country

In [8]:
%%sql SELECT
    d.time_id,
    d.cases,
    dt.date,
    m.category,
    m.value
FROM Data d
JOIN Dimtime dt
    ON d.time_id = dt.id
JOIN Measure m
    ON d.country_id = m.country_id
WHERE d.country_id = 201
LIMIT 500;

 * sqlite:///database.db
Done.


[(758, 0, '2019-12-31', 'Movement restrictions', 'Visa restrictions'),
 (759, 0, '2020-01-01', 'Movement restrictions', 'Visa restrictions'),
 (760, 0, '2020-01-02', 'Movement restrictions', 'Visa restrictions'),
 (761, 0, '2020-01-03', 'Movement restrictions', 'Visa restrictions'),
 (762, 0, '2020-01-04', 'Movement restrictions', 'Visa restrictions'),
 (763, 0, '2020-01-05', 'Movement restrictions', 'Visa restrictions'),
 (764, 0, '2020-01-06', 'Movement restrictions', 'Visa restrictions'),
 (765, 0, '2020-01-07', 'Movement restrictions', 'Visa restrictions'),
 (766, 0, '2020-01-08', 'Movement restrictions', 'Visa restrictions'),
 (767, 0, '2020-01-09', 'Movement restrictions', 'Visa restrictions'),
 (768, 0, '2020-01-10', 'Movement restrictions', 'Visa restrictions'),
 (769, 0, '2020-01-11', 'Movement restrictions', 'Visa restrictions'),
 (770, 0, '2020-01-12', 'Movement restrictions', 'Visa restrictions'),
 (771, 0, '2020-01-13', 'Movement restrictions', 'Visa restrictions'),
 (772, 0, '2020-01-14', 'Movement restrictions', 'Visa restrictions'),
 (773, 0, '2020-01-15', 'Movement restrictions', 'Visa restrictions'),
 (774, 0, '2020-01-16', 'Movement restrictions', 'Visa restrictions'),
 (775, 0, '2020-01-17', 'Movement restrictions', 'Visa restrictions'),
 (776, 0, '2020-01-18', 'Movement restrictions', 'Visa restrictions'),
 (777, 0, '2020-01-19', 'Movement restrictions', 'Visa restrictions'),
 (778, 0, '2020-01-20', 'Movement restrictions', 'Visa restrictions'),
 (779, 1, '2020-01-21', 'Movement restrictions', 'Visa restrictions'),
 (780, 0, '2020-01-22', 'Movement restrictions', 'Visa restrictions'),
 (781, 0, '2020-01-23', 'Movement restrictions', 'Visa restrictions'),
 (782, 0, '2020-01-24', 'Movement restrictions', 'Visa restrictions'),
 (783, 1, '2020-01-25', 'Movement restrictions', 'Visa restrictions'),
 (784, 0, '2020-01-26', 'Movement restrictions', 'Visa restrictions'),
 (785, 3, '2020-01-27', 'Movement restrictions', 'Visa restrictions'),
 (786, 0, '2020-01-28', 'Movement restrictions', 'Visa restrictions'),
 (787, 0, '2020-01-29', 'Movement restrictions', 'Visa restrictions'),
 (788, 0, '2020-01-30', 'Movement restrictions', 'Visa restrictions'),
 (789, 1, '2020-01-31', 'Movement restrictions', 'Visa restrictions'),
 (790, 1, '2020-02-01', 'Movement restrictions', 'Visa restrictions'),
 (791, 1, '2020-02-02', 'Movement restrictions', 'Visa restrictions'),
 (792, 3, '2020-02-03', 'Movement restrictions', 'Visa restrictions'),
 (793, 0, '2020-02-04', 'Movement restrictions', 'Visa restrictions'),
 (794, 0, '2020-02-05', 'Movement restrictions', 'Visa restrictions'),
 (795, 1, '2020-02-06', 'Movement restrictions', 'Visa restrictions'),
 (796, 0, '2020-02-07', 'Movement restrictions', 'Visa restrictions'),
 (797, 0, '2020-02-08', 'Movement restrictions', 'Visa restrictions'),
 (798, 0, '2020-02-09', 'Movement restrictions', 'Visa restrictions'),
 (799, 0, '2020-02-10', 'Movement restrictions', 'Visa restrictions'),
 (800, 1, '2020-02-11', 'Movement restrictions', 'Visa restrictions'),
 (801, 0, '2020-02-12', 'Movement restrictions', 'Visa restrictions'),
 (802, 1, '2020-02-13', 'Movement restrictions', 'Visa restrictions'),
 (803, 1, '2020-02-14', 'Movement restrictions', 'Visa restrictions'),
 (804, 0, '2020-02-15', 'Movement restrictions', 'Visa restrictions'),
 (805, 0, '2020-02-16', 'Movement restrictions', 'Visa restrictions'),
 (806, 0, '2020-02-17', 'Movement restrictions', 'Visa restrictions'),
 (807, 0, '2020-02-18', 'Movement restrictions', 'Visa restrictions'),
 (808, 0, '2020-02-19', 'Movement restrictions', 'Visa restrictions'),
 (809, 0, '2020-02-20', 'Movement restrictions', 'Visa restrictions'),
 (810, 1, '2020-02-21', 'Movement restrictions', 'Visa restrictions'),
 (811, 19, '2020-02-22', 'Movement restrictions', 'Visa restrictions'),
 (812, 0, '2020-02-23', 'Movement restrictions', 'Visa restrictions'),
 (813, 0, '2020-02-24', 'Movement restrictions', 'Visa restrictions'),
 (814, 18, '2020-02-25'

### Analysing country with the highest recovery cases (country id = 28)

#### Total number of cases in the country 

In [18]:
%%sql SELECT country_id, cumulate_cases
FROM data
WHERE country_id = 28
ORDER BY cumulate_cases DESC
LIMIT 1;

 * sqlite:///database.db
Done.


[(28, 1344143)]

#### Total number of recovery cases in the country

In [19]:
%%sql SELECT country_id, cumulate_recovered
FROM data
WHERE country_id = 28
ORDER BY cumulate_recovered DESC
LIMIT 1;

 * sqlite:///database.db
Done.


[(28, 746018)]

The recovery rate for the country is 55.5% 

#### Covid-19 transmission and protective measures taken by the country

In [9]:
%%sql SELECT
    d.time_id,
    d.cases,
    d.recovered,
    dt.date,
    m.category,
    m.value
FROM Data d
JOIN Dimtime dt
    ON d.time_id = dt.id
JOIN Measure m
    ON d.country_id = m.country_id
WHERE d.country_id = 28
LIMIT 400;

 * sqlite:///database.db
Done.


[(758, 0, None, '2019-12-31', 'Public health measures', 'Health screenings in airports and border crossings'),
 (759, 0, None, '2020-01-01', 'Public health measures', 'Health screenings in airports and border crossings'),
 (760, 0, None, '2020-01-02', 'Public health measures', 'Health screenings in airports and border crossings'),
 (761, 0, None, '2020-01-03', 'Public health measures', 'Health screenings in airports and border crossings'),
 (762, 0, None, '2020-01-04', 'Public health measures', 'Health screenings in airports and border crossings'),
 (763, 0, None, '2020-01-05', 'Public health measures', 'Health screenings in airports and border crossings'),
 (764, 0, None, '2020-01-06', 'Public health measures', 'Health screenings in airports and border crossings'),
 (765, 0, None, '2020-01-07', 'Public health measures', 'Health screenings in airports and border crossings'),
 (766, 0, None, '2020-01-08', 'Public health measures', 'Health screenings in airports and border crossings'),
 (767, 0, None, '2020-01-09', 'Public health measures', 'Health screenings in airports and border crossings'),
 (768, 0, None, '2020-01-10', 'Public health measures', 'Health screenings in airports and border crossings'),
 (769, 0, None, '2020-01-11', 'Public health measures', 'Health screenings in airports and border crossings'),
 (770, 0, None, '2020-01-12', 'Public health measures', 'Health screenings in airports and border crossings'),
 (771, 0, None, '2020-01-13', 'Public health measures', 'Health screenings in airports and border crossings'),
 (772, 0, None, '2020-01-14', 'Public health measures', 'Health screenings in airports and border crossings'),
 (773, 0, None, '2020-01-15', 'Public health measures', 'Health screenings in airports and border crossings'),
 (774, 0, None, '2020-01-16', 'Public health measures', 'Health screenings in airports and border crossings'),
 (775, 0, None, '2020-01-17', 'Public health measures', 'Health screenings in airports and border crossings'),
 (776, 0, None, '2020-01-18', 'Public health measures', 'Health screenings in airports and border crossings'),
 (777, 0, None, '2020-01-19', 'Public health measures', 'Health screenings in airports and border crossings'),
 (778, 0, None, '2020-01-20', 'Public health measures', 'Health screenings in airports and border crossings'),
 (779, 0, None, '2020-01-21', 'Public health measures', 'Health screenings in airports and border crossings'),
 (780, 0, None, '2020-01-22', 'Public health measures', 'Health screenings in airports and border crossings'),
 (781, 0, None, '2020-01-23', 'Public health measures', 'Health screenings in airports and border crossings'),
 (782, 0, None, '2020-01-24', 'Public health measures', 'Health screenings in airports and border crossings'),
 (783, 0, None, '2020-01-25', 'Public health measures', 'Health screenings in airports and border crossings'),
 (784, 0, None, '2020-01-26', 'Public health measures', 'Health screenings in airports and border crossings'),
 (785, 0, None, '2020-01-27', 'Public health measures', 'Health screenings in airports and border crossings'),
 (786, 0, None, '2020-01-28', 'Public health measures', 'Health screenings in airports and border crossings'),
 (787, 0, None, '2020-01-29', 'Public health measures', 'Health screenings in airports and border crossings'),
 (788, 0, None, '2020-01-30', 'Public health measures', 'Health screenings in airports and border crossings'),
 (789, 0, None, '2020-01-31', 'Public health measures', 'Health screenings in airports and border crossings'),
 (790, 0, None, '2020-02-01', 'Public health measures', 'Health screenings in airports and border crossings'),
 (791, 0, None, '2020-02-02', 'Public health measures', 'Health screenings in airports and border crossings'),
 (792, 0, None, '2020-02-03', 'Public health measures', 'Health screenings in airports and border crossings'),
 (793, 0, None, '2020-02-04', 'Public health measures', 'Health screenings in airports and border crossings'),
 (79

## Main findings from the project

1. Country id 201 has the highest covid-19 cases and deaths
2. Country id 28 has the highest recovery cases with a recovery rate of 55.5%
3. The total number of cases for year 2019 and year 2020 across 210 countries is 3,0002,042
4. Country id 201 has the highest is due to the slow action taken by the country; only implied visa restriction during the early stages of covid-19 transmission. This might allow people from unrestricted countries to cause transmision in the local community 
5. In contary to country id 28, which has the highest recovery rate. During the early covid-19 stages, the country implied health screenings in airports and border crossing which prevents covid-19 transmission from people who just returned from other countries